lib/Backends/OpenCL/OpenCLDeviceManager.cpp

```diff
@@ -64,6 +64,82 @@ static llvm::Expected<unsigned> parseInputAsUnsigned(std::string input) {
   return parsed;
 }
 
+OpenCLCommandQueuePool::~OpenCLCommandQueuePool() {
+  // Make sure all queues have been returned to the pool.
+  DCHECK_EQ(queuesAllocated_, queuesAvailable_)
+      << "OpenCLCommandQueue destroyed before all queues returned!";
+  // For each properties -> vector of queues pair:
+  for (auto &kv : queues_) {
+    // For each queue in each vector:
+    for (auto &q : kv.second) {
+      // Release the backing queue.
+      cl_int err = clReleaseCommandQueue(q.backingQueue);
+      DCHECK_EQ(err, CL_SUCCESS)
+          << "clReleaseCommandQueue failed with error code " << err;
+      // Adjust counters.
```

Comment:
The counters are destroyed at the exit of this function, do you need to adjust them?